src/main/java/factory/Logger.java

```diff
@@ -0,0 +1,28 @@
+package factory;
+
+public class Logger {
+    private static Logger logger;
+
+    private Logger() {
+
+    }
+
+    public static Logger getInstance(){
+        if(logger == null){
+            logger = new Logger();
+        }
+        return logger;
+    }
+
+    public void error() {
```

Comment:
normal behavior of a logger is to receive a message to log, so it'll be ideal to receive the message to report with that level of log